openedx/core/djangoapps/content_tagging/rest_api/v1/views.py

```diff
@@ -130,8 +146,120 @@ def orgs(self, request, **_kwargs) -> Response:
 class ObjectTagOrgView(ObjectTagView):
     """
     View to create and retrieve ObjectTags for a provided Object ID (object_id).
-    This view extends the ObjectTagView to add Organization filters for the results.
+    This view extends the ObjectTagView to add Organization filters for the results and
+    new actions like: export.
 
     Refer to ObjectTagView docstring for usage details.
     """
     filter_backends = [ObjectTagTaxonomyOrgFilterBackend]
+
+
+class ObjectTagExportView(APIView):
+    """"
+    View to export a CSV with all children and tags for a given object_id.
+    """
+    def get(self, request: Request, **kwargs) -> StreamingHttpResponse:
+        """
+        Export a CSV with all children and tags for a given object_id.
+        """
+
+        def _content_generator(
+            tagged_content: TaggedContent, level: int = 0
+        ) -> Iterator[tuple[TaggedContent, int]]:
+            """
+            Generator that yields the tagged content and the level of the block
+            """
+            yield tagged_content, level
+            if tagged_content.children:
+                for child in tagged_content.children:
+                    yield from _content_generator(child, level + 1)
+
+        class Echo(object):
+            """
+            Class that implements just the write method of the file-like interface,
+            used for the streaming response.
+            """
+            def write(self, value):
+                return value
+
+        def _generate_csv_rows(
+            tagged_content: TaggedContent,
+            taxonomies: TaxonomyDict,
+            pseudo_buffer: Echo,
+        ) -> Iterator[str]:
+            """
+            Receives the blocks, tags and taxonomies and returns a CSV string
+            """
+
+            header = {"name": "Name", "type": "Type", "id": "ID"}
+
+            # Prepare the header for the taxonomies
+            for taxonomy_id, taxonomy in taxonomies.items():
+                # ToDo: change to taxonomy.external_id after the external_id is implemented
+                header[f"taxonomy_{taxonomy_id}"] = taxonomy.name
+
+            csv_writer = csv.DictWriter(pseudo_buffer, fieldnames=header.keys())
+            yield csv_writer.writerow(header)
+
+            # Iterate over the blocks and yield the rows
+            for item, level in _content_generator(tagged_content):
+                if item.xblock.category == "course":
+                    block_id = item.xblock.id
+                else:
+                    block_id = item.xblock.location
+
+                block_data = {
+                    "name": level * "  " + item.xblock.display_name_with_default,
```

Comment:
You'll have to force-quote this field in CSV so that the prefixing spaces are included in the field data -- currently when I export a CSV and import it into Google Sheets, these spaces get ignored.

Reply:
I didn't find a way to force quoting a single column or value without ending with a literal quote in the string.
I changed the quoting to QUOTE_NONNUMERIC: 1f11b17e50e9ea94853c2d188fcbc10fe8d2f5bd
It adds a bit more quotes, but it will work.